### PR TITLE
Mule 19545

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/processor/ObjectTypeVisitor.java [hunk 1]
```diff
@@ -65,17 +65,16 @@ public void onType(Class<?> type) {
   @Override
   public void onConfigurationAttribute(String attributeName, Class<?> enforcedClass) {
     try {
-      type =
-          ClassUtils.getClass(Thread.currentThread().getContextClassLoader(),
-                              componentModel.getRawParameterValue(attributeName).orElse(null));
+      type = ClassUtils.getClass(Thread.currentThread().getContextClassLoader(),
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/processor/ObjectTypeVisitor.java:68]: static import

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/BeanDefinitionFactory.java [hunk 2]
```diff
@@ -650,15 +650,13 @@ private void processMuleConfiguration(Map<ComponentAst, SpringComponentModel> sp
   private BeanDefinitionCreator<CreateComponentBeanDefinitionRequest> buildComponentProcessorChainOfResponsability() {
     EagerObjectCreator eagerObjectCreator = new EagerObjectCreator();
     ObjectBeanDefinitionCreator objectBeanDefinitionCreator = new ObjectBeanDefinitionCreator();
-    PropertiesMapBeanDefinitionCreator propertiesMapBeanDefinitionCreator = new PropertiesMapBeanDefinitionCreator();
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/BeanDefinitionFactory.java:653]: image

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/ObjectBeanDefinitionCreator.java [hunk 3]
```diff
@@ -43,8 +43,10 @@ boolean handleRequest(Map<ComponentAst, SpringComponentModel> springComponentMod
     if (component == null || !component.getIdentifier().equals(buildFromStringRepresentation("mule:object"))) {
       return false;
     }
-    String refParameterValue = component.getParameter(REF_PARAMETER).getResolvedRawValue();
-    String classParameterValue = component.getParameter(CLASS_PARAMETER).getResolvedRawValue();
+    String refParameterValue =
+        component.getParameter(REF_PARAMETER) != null ? component.getParameter(REF_PARAMETER).getResolvedRawValue() : null;
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/ObjectBeanDefinitionCreator.java:47]: avoid the double lookup of the parameter

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/ObjectBeanDefinitionCreator.java [hunk 4]
```diff
@@ -43,8 +43,10 @@ boolean handleRequest(Map<ComponentAst, SpringComponentModel> springComponentMod
...
+    String classParameterValue =
+        component.getParameter(CLASS_PARAMETER) != null ? component.getParameter(CLASS_PARAMETER).getResolvedRawValue() : null;
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/ObjectBeanDefinitionCreator.java:49]: avoid the double lookup of the parameter

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/PropertyComponentUtils.java [hunk 5]
```diff
@@ -41,37 +41,24 @@
...
-            propertiesMap.put(mapEntry.getRawParameterValue(PROPERTY_NAME_MULE_PROPERTY_ATTRIBUTE).orElse(null), value);
-          });
-          return new Pair<>(propertyComponentModel.getComponentId().orElse(null), (Object) propertiesMap);
-        })
-        .orElseGet(() -> {
-          Object value = propertyComponentModel.getRawParameterValue(refKey)
-              .map(v -> (Object) new RuntimeBeanReference(v))
-              .orElseGet(() -> propertyComponentModel.getRawParameterValue(VALUE_PARAMETER_NAME)
-                  .orElse(null));
+    Object value;
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/PropertyComponentUtils.java:44]: is this method still needed at all?

**Review by elrodro83 — CHANGES_REQUESTED** [review]
See if `CommonComponentBeanDefinitionCreator#processMuleProperties` can also be removed

**Comment by mbuchwald** [comment]
--validate

**Review by elrodro83 — APPROVED** [review]

**Review by elrodro83 — APPROVED** [review]

File: tests/allure/src/main/java/org/mule/test/allure/AllureConstants.java [hunk 6]
```diff
@@ -862,6 +862,7 @@
 
   }
 
+<<<<<<< HEAD
```
Comment (elrodro83) [inline tests/allure/src/main/java/org/mule/test/allure/AllureConstants.java:865]: !!!!

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: tests/allure/src/main/java/org/mule/test/allure/AllureConstants.java (hunk 6 above)
Comment (mbuchwald) [reply tests/allure/src/main/java/org/mule/test/allure/AllureConstants.java:865]: Uh la re con...

**Review by elrodro83 — APPROVED** [review]

**Comment by mbuchwald** [comment]
--validate

**Pull request merged**